shell/src/main/java/alluxio/cli/docgen/ConfigurationDocGenerator.java

```diff
@@ -119,9 +155,19 @@ public static void writeCSVFile(Collection<? extends PropertyKey> defaultKeys, S
     } finally {
       try {
         closer.close();
+        if (validate) {
+          compareFiles(CSV_FILE_NAMES, filePath, "CSV");
+        }
       } catch (IOException e) {
         LOG.error("Error while flushing/closing Property Key CSV FileWriter", e);
       }
+      for (String fileName : CSV_FILE_NAMES) {
```

Comment:
the following deletion operation only needs to occur if validate is true. if validate is false, the temp files are known not to exist.

same for the yml files iteration